core/src/main/scala/swave/core/StreamOps.scala

```diff
@@ -312,6 +312,46 @@ trait StreamOps[A] extends Any { self ⇒
   final def slice(startIndex: Long, length: Long): Repr[A] =
     via(Pipe[A] drop startIndex take length named "slice")
 
+  final def sliding(windowSize: Int, emitSingleEmpty: Boolean = false): Repr[immutable.Seq[A]] =
+    slidingTo[immutable.Seq](windowSize, emitSingleEmpty)
+
+  final def slidingTo[M[+ _]](windowSize: Int, emitSingleEmpty: Boolean = false)(
+      implicit cbf: CanBuildFrom[M[A], A, M[A]]): Repr[M[A]] = {
+
+    val builder = cbf.apply()
+
+    prefixAndTail(windowSize).map {
```

Comment:
Yes, you are right. I see now where the problem is with the approach I had in mind before.
`prefixAndTail` indeed appears like the cleanest solution!